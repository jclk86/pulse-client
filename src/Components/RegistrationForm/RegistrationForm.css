.bg_mobile_view_registration_page {
  display: none;
}

.container_desktop_view_registration_hero {
  display: none;
}

.logo_registration {
  width: 150px;
  margin: 0 auto;
}

#header_registration_form {
  margin-top: 0;
  color: black;
}

.RegistrationForm {
  padding-bottom: 30px;
  max-width: 600px;
  background-image: url("../../images/main_bg.jpeg");
}

.container_RegistrationForm_username,
.container_RegistrationForm_password,
.container_RegistrationForm_profile,
.container_RegistrationForm_email,
.container_RegistrationForm_image_url,
.container_RegistrationForm_fullname {
  width: 80%;
  margin: 10px auto;
}

#label_registration_fullname,
#label_registration_username,
#label_registration_password,
#label_registration_email,
#label_registration_profile,
#label_registration_image_url {
  display: block;
  width: 100%;
  text-align: left;
  font-size: 16px;
  font-weight: bold;
  color: black;
}

.RegistrationForm_email,
.RegistrationForm_username,
.RegistrationForm_image_url,
.RegistrationForm_password,
.RegistrationForm_fullname {
  padding: 10px;
  width: 100%;
  margin-top: 10px;
  border-radius: 3px;
  border: 1px solid transparent;
  border-top: none;
  border-bottom: 1px solid #ddd;
  box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.39), 0 -1px 1px #fff, 0 1px 0 #fff;
  font-size: 16px;
}

.RegistrationForm_email:focus,
.RegistrationForm_username:focus,
.RegistrationForm_image_url:focus,
.RegistrationForm_password:focus,
.RegistrationForm_fullname:focus,
.RegistrationForm_profile:focus {
  outline-width: 0;
}

.RegistrationForm_email:hover,
.RegistrationForm_username:hover,
.RegistrationForm_image_url:hover,
.RegistrationForm_password:hover,
.RegistrationForm_fullname:hover,
.RegistrationForm_profile:hover {
  border: 1px solid rgb(110, 110, 110);
}

.RegistrationForm_profile {
  display: block;
  font-size: 16px;
  padding: 10px;
  margin: 10px auto;
  text-align: left;
  width: 100%;
  border-radius: 10px;
  font-family: "Open Sans", sans-serif;
  resize: none;
  min-height: 150px;
  border-radius: 3px;
  border: 1px solid transparent;
  border-top: none;
  border-bottom: 1px solid #ddd;
  box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.39), 0 -1px 1px #fff, 0 1px 0 #fff;
}

.RegistrationForm_submit_btn {
  min-width: 200px;
  margin-top: 10px;
  padding: 10px;
  border-radius: 10px;
  font-size: 16px;
  background-color: white;
  color: black;
  font-weight: bold;
  border: 1px solid transparent;
  border-top: none;
  border-bottom: 1px solid #ddd;
  box-shadow: 2px 2px 5px;
}

.RegistrationForm_submit_btn:hover {
  cursor: pointer;
  border: 1px solid rgb(110, 110, 110);
}

.RegistrationForm_login_link {
  text-decoration: none;
  color: teal;
}

.red {
  margin-top: 2px;
  color: rgb(255, 0, 0);
}

#message_redirect {
  color: black;
}
@media only screen and (min-width: 1024px) {
  .container_desktop_view_registration_hero {
    display: block;
    font-size: 30px;
    padding-top: 250px;
    width: 75%;
    margin: 0 auto;
  }

  .container_registration_page {
    display: flex;
    min-height: 100vh;
  }

  .container_registration_form {
    width: 30%;
    height: 100%;
  }

  .RegistrationForm {
    background-image: none;
    background-color: rgb(211, 211, 211);
    display: flex;
    flex-direction: column;
  }

  .bg_desktop_view_registration_page {
    background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)),
      url("../../images/celebration.jpeg") no-repeat center center fixed;
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
    min-height: 100vh;
    overflow: auto;
    width: 70%;
  }
}
