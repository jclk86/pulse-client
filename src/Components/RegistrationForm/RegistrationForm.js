import React, { Component } from "react";
import { NavLink } from "react-router-dom";
import { Form } from "../Utils/Utils";
import AuthApiService from "../../Services/auth-api-service";

class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  };

  constructor(props) {
    super(props);
    this.state = {
      error: null,
      password: { value: "", touched: false },
      email: { value: "", touched: false },
      username: { value: "", touched: false },
      fullname: { value: "", touched: false }
    };
  }

  updateFullname = fullname => {
    this.setState({ fullname: { value: fullname, touched: true } });
  };

  updateUsername = username => {
    this.setState({ username: { value: username, touched: true } });
  };

  updatePassword = password => {
    this.setState({ password: { value: password, touched: true } });
  };

  updateEmail = email => {
    this.setState({ email: { value: email, touched: true } });
  };

  handleSubmit = event => {
    event.preventDefault();
    const { fullname, username, password, email } = event.target;
    this.setState({
      error: null,
      password: password,
      email: email,
      username: username,
      fullname: fullname
    });

    AuthApiService.postUser({
      fullname: fullname.value,
      username: username.value,
      password: password.value,
      email: email.value
    })
      .then(res => {
        fullname.value = "";
        username.value = "";
        password.value = "";
        email.value = "";
        this.props.onRegistrationSuccess();
      })
      .catch(res => {
        this.setState({ error: res.error });
      });
  };

  render() {
    return (
      <Form onSubmit={this.handleSubmit} id="registration_form">
        <div className="fullname">
          <label
            htmlFor="RegistrationForm__fullname"
            className="label_registration"
          >
            Fullname
          </label>
          <input
            name="fullname"
            type="text"
            required
            id="RegistrationForm__fullname"
            onChange={e => this.updateFullname(e.target.value)}
          />
        </div>
        <div className="user_name">
          <label
            htmlFor="RegistrationForm__username"
            className="label_registration"
          >
            Username
          </label>
          <input
            name="username"
            type="text"
            required
            id="RegistrationForm__username"
            onChange={e => this.updateUsername(e.target.value)}
          />
        </div>
        <div className="password">
          <label
            htmlFor="RegistrationForm__password"
            className="label_registration"
          >
            Password
          </label>
          <input
            onChange={e => this.updatePassword(e.target.value)}
            name="password"
            type="password"
            id="RegistrationForm__password"
          />
        </div>
        <div className="email">
          <label
            htmlFor="RegistrationForm__email"
            className="label_registration"
          >
            Email
          </label>{" "}
          <input
            onChange={e => this.updateEmail(e.target.value)}
            name="email"
            type="text"
            required
            id="RegistrationForm__email"
          />
        </div>
        <div className="register_btn_container">
          <button id="submit_btn" type="submit">
            Submit
          </button>
        </div>
        <div className="login_link">
          <p className="message_redirect">
            Already a user?{" "}
            <NavLink to={"/login"} role="navigation" className="btn_login_link">
              Login
            </NavLink>
          </p>
        </div>
      </Form>
    );
  }
}

export default RegistrationForm;
