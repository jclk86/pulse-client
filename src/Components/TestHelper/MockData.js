export const mockUser = {
  username: "Pop123",
  password: "Password123!",
  fullname: "Bob Smith",
  email: "BSmith@gmail.com"
};

export const mockArticle = {
  title: "test article",
  content: "test content",
  article_tag: "News",
  id: 1,
  date_created: "2019-10-14T21:49:20.570Z",
  author: {
    username: mockUser.username
  }
};

export const mockComment = {
  id: 1,
  content: "test comment content",
  date_created: "2019-10-14T21:49:20.570Z",
  article_id: 1,
  user: {
    username: mockArticle.author.username
  }
};
