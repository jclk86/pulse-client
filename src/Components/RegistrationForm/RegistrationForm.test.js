import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RegistrationForm from "./RegistrationForm";
import sinon from "sinon";
import { mockUser } from "../TestHelper/MockData";

describe("Registration Form Component", () => {
  it("renders registration form without issues", () => {
    const wrapper = shallow(<RegistrationForm></RegistrationForm>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("fires submit", () => {
    const onSubmit = sinon.spy();
    const wrapper = mount(
      <BrowserRouter>
        <RegistrationForm onSubmit={onSubmit}></RegistrationForm>
      </BrowserRouter>
    );
    wrapper.find("Form").simulate("submit", {
      target: {
        username: mockUser.username,
        password: mockUser.password,
        fullname: mockUser.fullname,
        email: mockUser.email,
        profile: mockUser.profile,
        image_url: mockUser.image_url
      },
      preventDefault: () => {}
    });
  });
});
