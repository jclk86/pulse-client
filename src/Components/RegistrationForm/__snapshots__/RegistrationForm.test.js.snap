// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Form Component renders registration form without issues 1`] = `
<div
  className="container_registration_page"
>
  <div
    className="container_registration_form"
  >
    <Form
      className="RegistrationForm"
      id="no_radius"
      onSubmit={[Function]}
    >
      <Logo
        className="logo_registration"
      />
      <h2
        id="header_registration_form"
      >
        Registration
      </h2>
      <div
        className="container_RegistrationForm_fullname"
      >
        <label
          htmlFor="RegistrationForm_fullname"
          id="label_registration_fullname"
        >
          Fullname
        </label>
        <Input
          className="RegistrationForm_fullname"
          name="fullname"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="container_RegistrationForm_username"
      >
        <label
          htmlFor="RegistrationForm_username"
          id="label_registration_username"
        >
          Username
        </label>
        <Input
          className="RegistrationForm_username"
          name="username"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="container_RegistrationForm_password"
      >
        <label
          htmlFor="RegistrationForm_password"
          id="label_registration_password"
        >
          Password
        </label>
        <Input
          autoComplete="off"
          className="RegistrationForm_password"
          name="password"
          onChange={[Function]}
          type="password"
        />
      </div>
      <div
        className="container_RegistrationForm_email"
      >
        <label
          htmlFor="RegistrationForm_email"
          id="label_registration_email"
        >
          Email
        </label>
         
        <Input
          className="RegistrationForm_email"
          name="email"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="container_RegistrationForm_profile"
      >
        <label
          htmlFor="RegistrationForm_profile"
          id="label_registration_profile"
        >
          About Me
        </label>
         
        <textarea
          className="RegistrationForm_profile"
          name="profile"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="container_RegistrationForm_image_url"
      >
        <label
          htmlFor="RegistrationForm_image_url"
          id="label_registration_image_url"
        >
          Image Url
        </label>
         
        <Input
          className="RegistrationForm_image_url"
          name="image_url"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        role="alert"
      />
      <div
        className="container_RegistrationForm_btn"
      >
        <button
          className="RegistrationForm_submit_btn"
          disabled={true}
          type="submit"
        >
          Submit
        </button>
      </div>
      <div
        className="container_RegistrationForm_login_link"
      >
        <p
          id="message_redirect"
        >
          Already a user?
           
          <NavLink
            className="RegistrationForm_login_link"
            to="/login"
          >
            Login
          </NavLink>
        </p>
      </div>
    </Form>
  </div>
  <div
    className="bg_desktop_view_registration_page"
  >
     
    <div
      className="container_desktop_view_registration_hero"
    >
      <h3
        role="banner"
      >
        Become a part of our community with a simple sign-up for a free account.
      </h3>
    </div>
  </div>
</div>
`;
